packages/webapp/src/pages/EnvironmentSettings.tsx

```diff
@@ -616,9 +596,9 @@ export const EnvironmentSettings: React.FC = () => {
                                 </label>
                             </div>
                             <div className="">
-                                <Button className="items-center" variant="primary" onClick={slackIsConnected ? disconnectSlack : connectSlack}>
-                                    <IntegrationLogo provider="slack" height={6} width={6} classNames="" />
-                                    {slackIsConnected ? 'Disconnect' : 'Connect'}
+                                <Button className="items-center" variant="primary" onClick={slackIsConnected ? disconnectSlack : createSlackConnection}>
+                                    <IntegrationLogo provider="slack" height={5} width={6} classNames="" />
+                                    {slackIsConnected ? `Disconnect ${slackConnectedChannel}` : 'Connect'}
                                 </Button>
```

Comment:
Right now this button fails silently if you don't have a Slack provider setup. OnFailure should have at least displayed a toast (which didn't) but a message or a disabled button would be even better

Reply:
Updated with https://github.com/NangoHQ/nango/pull/2184/commits/67ac5a6bf7b8f9fbfc99626c38bc06b25a30a1b0